pkg/api/datasources.go

```diff
@@ -351,28 +352,57 @@ func validateJSONData(ctx context.Context, jsonData *simplejson.Json, cfg *setti
 
 	// Prevent adding a data source team header with a name that matches the auth proxy header name
 	if features.IsEnabled(featuremgmt.FlagTeamHttpHeaders) {
-		teamHTTPHeadersJSON, err := datasources.GetTeamHTTPHeaders(jsonData)
+		err := validateTeamHTTPHeaderJSON(jsonData)
 		if err != nil {
-			datasourcesLogger.Error("Unable to marshal TeamHTTPHeaders")
-			return errors.New("validation error, invalid format of TeamHTTPHeaders")
+			return err
 		}
-		// whitelisting X-Prom-Label-Policy
-		for _, headers := range teamHTTPHeadersJSON {
-			for _, header := range headers {
-				// TODO: currently we only allow for X-Prom-Label-Policy header to be used by our proxy
-				for _, name := range []string{"X-Prom-Label-Policy"} {
-					if http.CanonicalHeaderKey(header.Header) != http.CanonicalHeaderKey(name) {
-						datasourcesLogger.Error("Cannot add a data source team header that is different than", "headerName", name)
-						return errors.New("validation error, invalid header name specified")
-					}
+	}
+
+	return nil
+}
+
+func validateTeamHTTPHeaderJSON(jsonData *simplejson.Json) error {
+	teamHTTPHeadersJSON, err := datasources.GetTeamHTTPHeaders(jsonData)
+	if err != nil {
+		datasourcesLogger.Error("Unable to marshal TeamHTTPHeaders")
+		return errors.New("validation error, invalid format of TeamHTTPHeaders")
+	}
+	// whitelisting ValidHeaders
+	for _, headers := range teamHTTPHeadersJSON {
+		for _, header := range headers {
+			for _, name := range ValidHeaders() {
+				if http.CanonicalHeaderKey(header.Header) != http.CanonicalHeaderKey(name) {
```

Comment:
There's a bit of a problem here. If you have multiple valid headers (ie foo and bar), then this code always return an error, since even if foo == foo, but foo != bar

Reply:
this is only checking the header name. i do not really understand this one 🤔

Reply:
we needed to allow for a slice of strings here.